### PR TITLE
Refactor more cluster creation code into NewCluster()

### PR DESCRIPTION
**Pull request by johngmyers**

### CODE REVIEW

**Comment by olemarkus** [comment]
/lgtm

Assume the renaming is to match the kops API

**Comment by johngmyers** [comment]
Yes, but I'm wondering what I should do about `OpenstackNetworkID`, which appears to be separate only so the flag can have a different name. Perhaps I should map that to `NetworkID` in create_cluster.

**Comment by olemarkus** [comment]
I'd map it. 

/cc @zetaab

**Comment by johngmyers** [comment]
I mapped it.

File: pkg/apis/kops/channel.go [hunk 1]
```diff
@@ -84,6 +84,10 @@ type ChannelImageSpec struct {
 
 // LoadChannel loads a Channel object from the specified VFS location
 func LoadChannel(location string) (*Channel, error) {
+	if location == "none" {
```
Comment (rifelpet) [inline pkg/apis/kops/channel.go:87]: why is this needed? I don't see `none` referenced anywhere else

Comment (johngmyers) [reply pkg/apis/kops/channel.go:87]: This allows for a simple way to opt out of channels.

Comment (rifelpet) [reply pkg/apis/kops/channel.go:87]: ah ok. It would be helpful to document the different values for `--channel` in the flag's help text (`stable`, `alpha`, `none`, or a URL? any others?) but that could be done in a separate PR.

Comment (rifelpet) [reply pkg/apis/kops/channel.go:87]: and consider adding `none` here: https://github.com/kubernetes/kops/blob/master/pkg/apis/kops/channel.go#L33 and using it in the string comparison. I'd be hesitant to add an `alpha` const because we don't actually reference the specific value anywhere, we just treat it as a relative path appended to the channel base URL. But maybe it would be helpful for the flag's help text.
